python/feathub/processors/flink/table_builder/over_window_utils.py

```diff
@@ -196,73 +135,71 @@ def _get_flink_over_window(
-    ):
-        interval_expr = native_flink_expr.call_sql(
-            timedelta_to_flink_sql_interval(
-                over_window_descriptor.window_size, day_precision=3
-            )
-        )
+    gateway = get_gateway()
+    j_feathub_agg = gateway.jvm.com.alibaba.feathub.flink.udf.aggregation
 
-        java_udf_descriptor = ROW_AND_TIME_BASED_OVER_WINDOW_JAVA_UDF.get(
-            agg_func, None
-        )
-        if java_udf_descriptor is None:
-            raise FeathubTransformationException(
-                f"Unsupported aggregation for FlinkProcessor {agg_func}."
-            )
+    agg_func = agg_descriptor.agg_func
+    j_agg_func = j_feathub_agg.AggFuncUtils.getAggFunc(
+        agg_func.value, _to_java_data_type(input_datatype), True
+    )
 
-        return native_flink_expr.call(
-            java_udf_descriptor.udf_name,
-            interval_expr,
-            expr,
-            native_flink_expr.col(EVENT_TIME_ATTRIBUTE_NAME),
-        ).over(native_flink_expr.col(window_alias))
+    agg_field_expr = native_flink_expr.call_sql(agg_descriptor.expr)
 
-    if agg_func == AggFunc.AVG:
-        result = expr.avg
-    elif agg_func == AggFunc.COUNT:
-        result = expr.count
-    elif agg_func == AggFunc.MIN:
-        result = expr.min
-    elif agg_func == AggFunc.MAX:
-        result = expr.max
-    elif agg_func == AggFunc.SUM:
-        result = expr.sum
-    elif agg_func == AggFunc.VALUE_COUNTS:
-        result = native_flink_expr.call("value_counts", expr)
-    elif agg_func == AggFunc.COLLECT_LIST:
-        result = native_flink_expr.call("collect_list", expr)
-    elif agg_func == AggFunc.FIRST_VALUE:
-        result = expr.first_value
-    elif agg_func == AggFunc.LAST_VALUE:
-        result = expr.last_value
-    else:
-        raise FeathubTransformationException(
-            f"Unsupported aggregation for FlinkProcessor {agg_func}."
+    if agg_descriptor.limit is not None:
+        if over_window_descriptor.window_size is not None:
+            j_agg_func = j_feathub_agg.AggFuncWithLimit(
+                j_agg_func, agg_descriptor.limit
+            )
+        else:
+            j_agg_func = j_feathub_agg.UnboundedOverWindowAggFuncWithLimit(
+                j_agg_func, agg_descriptor.limit
+            )
+
+    if agg_descriptor.filter_expr is not None:
+        j_agg_func = j_feathub_agg.AggFuncWithFilterFlag(j_agg_func)
+        agg_field_expr = native_flink_expr.row(
+            agg_field_expr,
+            native_flink_expr.call_sql(agg_descriptor.filter_expr),
         )
 
-    return result.over(native_flink_expr.col(window_alias))
+    j_agg_func = j_feathub_agg.AggFuncAdaptor(j_agg_func)
+    j_expr = gateway.jvm.org.apache.flink.table.api.Expressions.call(
+        j_agg_func,
+        to_jarray(
+            gateway.jvm.Object,
+            [
+                agg_field_expr._j_expr_or_property_name,
+                native_flink_expr.col(
+                    EVENT_TIME_ATTRIBUTE_NAME
```

Comment:
Why do we need to add the timestamp column to the aggregation function's parameter list? Could you provide some document reference?

Reply:
The `AggFunc#add` method requires the event timestamp of the value. Therefore, the `AggFuncAdaptor` that implements `AggregateFunction` requires event time.